flower/utils/results/stores/database.py

```diff
@@ -1,13 +1,107 @@
+            query_limit = limit
+
+        session: Session
+        with self.backend.ResultSession() as session:
+            ordering = TaskExtended.date_done.desc() if reverse else TaskExtended.date_done.asc()
+            task_select_query = sa.select(TaskExtended).order_by(ordering).limit(query_limit)
+            for task in session.execute(task_select_query).scalars():
+                result = self._map_task_to_result(task)
+                yield result.task_id, result
+
+    def _map_task_to_result(self, task: TaskExtended) -> Result:
+        """
+        Convert a `TaskExtended` ORM object into our shared `Result` data structure. This class assumes the usage of
+        `TaskExtended` in order to query the "taskmeta" table, since `TaskExtended` queries can successfully return
+        full data for both tasks that were saved with `result_extended=True` and those that were saved with
+        `result_extended=False`.
+
+        Because we want to support both extended and non-extended tasks, we need a way to figure out whether the
+        provided task was _actually_ extended or not at the time it was saved. We can do this by looking at the `name`
+        field. When a task is saved under `result_extended=True`, then it will have a name referencing the name of the
+        function. Otherwise, that field will be null and we know that it was `result_extended=False`.
+        """
+        is_actually_extended: bool = task.name is not None
+        if is_actually_extended:
+            return Result(
+                task_id=task.task_id,
+                status=task.status,
+                date_done=task.date_done,
+                result=task.result,
+                traceback=task.traceback,
+                args=self.deserialize_binary_column_value(task.args),
+                kwargs=self.deserialize_binary_column_value(task.kwargs),
+                name=task.name,
+            )
+
+        return Result(
+            task_id=task.task_id,
+            status=task.status,
+            date_done=task.date_done,
+            result=task.result,
+            traceback=task.traceback,
+        )
+
+    def deserialize_binary_column_value(self, value: bytes) -> Any:
+        """
+        Attempt to deserialize the provided `value` using the available serialization decoders. Celery stores task
+        `args` and `kwargs` in binary columns, but the proper decoding mechanism for those binary columns is not
+        immediately obvious. These fields get serialized bsed on whatever the value of `Celery.conf.result_serializer`
+        is at the time the task result is saved. However, it's possible that the value of that config setting changed
+        across different Celery processes, and therefore we may be dealing with a database that has co-mingled records
+        from different serializers. Unfortunately, there is no column in the database schema that records which
+        serializer was used for each task.
```

Comment:
sounds like somewhere there is missing a version/serial-number of what binary was stored, is there any chance we could add that binary and fail-out when our current bincompat number is larger than the recalled bincompat numbers?

postgres does something i think is smart and they only support the current and precious bincompat numbers, so as long as you run the service once on every release version, it will upgrade the stored data up one version.

Did this case get discovered in testing?

aside, again, red-flags all over this.

again semi-predicate problem here, which means we don't actually know what went wrong at runtime, which is another red flag

Reply:
I see your point about the semipredicate problem; we would want _some_ way to know that things are going wrong. Perhaps we could append some sort of `(Failed to decode)` prefix to the return value? That allows the user to know that an error took place, but they also have some visibility into the byte string in case it's useful. 

Re: serial number. We're basically at the mercy of `pickle`, `json`, or whatever data encoding gets used. We don't control how Celery translates its args/kwargs/results to Postgres binary objects upon save, so it's not easy for us to introduce versioning or any other type of behavior to the database table.